cob_twist_controller/src/inverse_jacobian_calculations/inverse_jacobian_calculation.cpp

```diff
@@ -76,18 +77,18 @@ Eigen::MatrixXd PInvBySVD::calculate(const TwistControllerParams& params,
         for (uint32_t i = 0; i < singularValues.rows()-1; ++i)
         {
             // pow(beta, 2) << pow(lambda, 2)
-            singularValuesInv(i) = singularValues(i) / (pow(singularValues(i), 2) + pow(params.beta, 2));
+            singularValuesInv(i) = singularValues(i) / (pow(singularValues(i), 2) + pow(params.beta, 2)+ lambda(i,i));
         }
         // Formula 20 - additional part - numerical filtering for least singular value m
-        uint32_t m = singularValues.rows();
-        singularValuesInv(m) = singularValues(m) / (pow(singularValues(m), 2) + pow(params.beta, 2) + pow(lambda, 2));
+        //uint32_t m = singularValues.rows();
+        //singularValuesInv(m) = singularValues(m) / (pow(singularValues(m), 2) + pow(params.beta, 2) + pow(lambda, 2));
```

Comment:
Yes, the way you handled ```pow(lambda,2)``` is ok...

But have a look at the indices of ```singularValuesInv(i)```!
In numerical_filtering, all indizes ```i=0,...,singularValues.rows()-1``` are __not__ damped, i.e. no ```+ lambda(i,i)``` and only index ```singularValues.rows()``` (which is the last/smallest singular value) is damped with lambda, i.e. ```+ lambda(i,i)```!